### PR TITLE
Aline patches

### PR DESCRIPTION
**Pull request by alvations**
## Summary

Fix IPA feature matrix errors, add input validation, add `vowels` list, and expand test coverage for the ALINE phonetic alignment algorithm. Attempt at patching https://github.com/nltk/nltk/issues/3523 and taking in good ideas from https://github.com/nltk/nltk/pull/3446

### Feature matrix fixes

| Vowel | Feature | Before | After | Reason |
|-------|---------|--------|-------|--------|
| ɯ | `round` | `"plus"` | `"minus"` | IPA ɯ is the close back **unrounded** vowel (unrounded counterpart of u) |
| ɘ | `high` | `"high"` | `"mid"` | IPA ɘ is a **close-mid** central vowel, consistent with peers e, o, ø, ɤ which all map to `"mid"` |
| ɵ | `high` | `"high"` | `"mid"` | IPA ɵ is a **close-mid** central rounded vowel (rounded counterpart of ɘ) |

These vowels were added to the NLTK port as IPA extensions (not in Kondrak's original). The bugs were introduced with incorrect feature assignments.

### Remove duplicate `"ʐ "` consonant entry

The consonants list contained both `"ʐ "` (with trailing space, 2 chars) and `"ʐ"` (without space, 1 char) since the very first commit (`91b90b6e5`). 

The `"ʐ "` entry was dead code: `align()` iterates char-by-char so a 2-character string can never match, and only `"ʐ"` (no space) exists in `feature_matrix`. It was a copy-paste artifact from
formatting the list across two lines.

### Add `vowels` list

The module had an explicit `consonants` list but no corresponding `vowels` list — vowels were only implicitly defined as `feature_matrix` entries with `place == "vowel"`. Added an explicit `vowels` list
for API parity.

### Add input validation

- `align()` now validates all input segments against `feature_matrix` at entry, raising `ValueError` with the segment and position instead of a cryptic `KeyError` deep in the algorithm
- Extracted `_validate_segments()` helper for reuse
- `sigma_exp()` now validates the expansion pair has exactly 2 segments

### Test coverage (28 tests, up from 6)

- Edge cases: identical strings, single char, single vowel, invalid segment error
- Feature matrix validation: all consonants/vowels in matrix, no trailing whitespace, place consistency, IPA rounding correctness, IPA height correctness
- Scoring functions: delta identity/symmetry/ordering, sigma_skip negativity, sigma_sub identity preference, sigma_exp, V() vowel vs consonant, R() return type
- Alignment properties: non-empty results, return type structure

### CODE REVIEW

File: nltk/metrics/aline.py [hunk 1]
```diff
@@ -102,7 +102,6 @@
     "ʃ",
     "ʈ",
     "ʋ",
-    "ʐ ",
```
Comment (alvations) [inline nltk/metrics/aline.py:105]: This is repeated down in L115

File: nltk/metrics/aline.py [hunk 2]
```diff
@@ -1206,7 +1245,7 @@
         "lateral": "minus",
         "high": "high",
         "back": "back",
-        "round": "plus",
+        "round": "minus",
```
Comment (alvations) [inline nltk/metrics/aline.py:1248]: ɯ is indeed non-rounded. https://en.wikipedia.org/wiki/Close_back_unrounded_vowel

File: nltk/metrics/aline.py [hunk 3]
```diff
@@ -1246,7 +1285,7 @@
         "nasal": "minus",
         "retroflex": "minus",
         "lateral": "minus",
-        "high": "high",
+        "high": "mid",
```
Comment (alvations) [inline nltk/metrics/aline.py:1288]: ɘ is indeed mid not high https://en.wikipedia.org/wiki/Close-mid_central_unrounded_vowel

File: nltk/metrics/aline.py [hunk 4]
```diff
@@ -1274,7 +1313,7 @@
...
```
Comment (alvations) [inline nltk/metrics/aline.py:1316]: ɵ is indeed mid https://en.wikipedia.org/wiki/Close-mid_central_rounded_vowel

**Comment by stevenbird** [comment]
Thanks @alvations
